File: dataset_reduzido/codenet/Python/Code/s205112814.py
n, m = map(int, input().split())

gate_cond = [[int(x) for x in input().split()] for _ in range(m)]
max_under_th = max([l[0] for l in gate_cond])
min_upper_th = min([l[1] for l in gate_cond])
if min_upper_th - max_under_th >= 0:
    print(min_upper_th - max_under_th + 1)
else:
    print(0)

File: output/StarCoder-100/codenet/Python/Java/s205112814.java
import java.util.*;

public class s205112814 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[] gate_cond = new int[m];
        for (int i = 0; i < m; i++) {
            gate_cond[i] = sc.nextInt();
        }
        int max_under_th = Integer.MIN_VALUE;
        for (int i = 0; i < m; i++) {
            max_under_th = Math.max(max_under_th, gate_cond[i][0]);
        }
        int min_upper_th = Integer.MAX_VALUE;
        for (int i = 0; i < m; i++) {
            min_upper_th = Math.min(min_upper_th, gate_cond[i][1]);
        }
        System.out.println(Math.max(0, min_upper_th - max_under_th + 1));
    }
}